src/backend/routers/announcements.py

```
@@ -0,0 +1,76 @@
+        "message": message,
+        "start_date": start_date,
+        "expiration_date": expiration_date,
+        "created_at": now,
+        "updated_at": now
+    }
+    result = announcements_collection.insert_one(ann)
+    return {"id": str(result.inserted_id), "message": "追加しました"}
+
+@router.put("/update/{announcement_id}")
+def update_announcement(announcement_id: str, message: Optional[str] = None, expiration_date: Optional[str] = None, start_date: Optional[str] = None, teacher_username: str = Query(...)):
+    """お知らせ編集（認証必須）"""
```

Ruff (0.14.8)

Docstring contains ambiguous `（` (FULLWIDTH LEFT PARENTHESIS). Did you mean `(` (LEFT PARENTHESIS)?

(RUF002)